doc/reference/models.md

```diff
@@ -0,0 +1,38 @@
+# Models reference
+
+
+## asset
```

Comment:
We should probably think a little bit about the order; e.g. Datasets before assets, layers before layer groups and steps? More generally, parent concepts before child concepts?